src/lib/pubkey/eckcdsa/eckcdsa.cpp

```diff
@@ -221,41 +219,41 @@ class ECKCDSA_Verification_Operation final : public PK_Ops::Verification {
       bool m_prefix_used;
 };
 
-void ECKCDSA_Verification_Operation::update(const uint8_t msg[], size_t msg_len) {
+void ECKCDSA_Verification_Operation::update(std::span<const uint8_t> msg) {
    if(!m_prefix_used) {
       m_prefix_used = true;
       m_hash->update(m_prefix.data(), m_prefix.size());
    }
-   m_hash->update(msg, msg_len);
+   m_hash->update(msg);
 }
 
-bool ECKCDSA_Verification_Operation::is_valid_signature(const uint8_t sig[], size_t sig_len) {
+bool ECKCDSA_Verification_Operation::is_valid_signature(std::span<const uint8_t> sig) {
    m_prefix_used = false;
-   secure_vector<uint8_t> digest = m_hash->final();
+   std::vector<uint8_t> digest = m_hash->final_stdvec();
    truncate_hash_if_needed(digest, m_group.get_order_bytes());
-   return verify(digest.data(), digest.size(), sig, sig_len);
+   return verify(digest, sig);
 }
 
-bool ECKCDSA_Verification_Operation::verify(const uint8_t msg[], size_t msg_len, const uint8_t sig[], size_t sig_len) {
+bool ECKCDSA_Verification_Operation::verify(std::span<const uint8_t> msg, std::span<const uint8_t> sig) {
    //calculate size of r
 
    const size_t order_bytes = m_group.get_order_bytes();
 
-   const size_t size_r = std::min(msg_len, order_bytes);
-   if(sig_len != size_r + order_bytes) {
+   const size_t size_r = std::min(msg.size(), order_bytes);
+   if(sig.size() != size_r + order_bytes) {
       return false;
    }
 
-   secure_vector<uint8_t> r(sig, sig + size_r);
+   std::vector<uint8_t> r(sig.begin(), sig.begin() + size_r);
```

Comment:
This could be a `sig.first(size_r)`, with some adaptions in the usage below. Most notably, the comparison with `v`.